lang/funcs/string.go

```diff
@@ -46,8 +46,85 @@ var ReplaceFunc = function.New(&function.Spec{
 	},
 })
 
+var SnakeCaseFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name:             "str",
+			Type:             cty.String,
+			AllowDynamicType: true,
+		},
+	},
+	Type: function.StaticReturnType(cty.String),
+	Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+		in := args[0].AsString()
+		// Remove non alpha-numerics
+		out := regexp.MustCompile(`(?m)[^a-zA-Z0-9]`).ReplaceAllString(in, "_")
+		// Split on uppercase characters followed by lower case (e.g. camel case)
+		out = regexp.MustCompile(`(?m)[A-Z][a-z]`).ReplaceAllString(out, "_$0")
+		// Remove any consecutive underscores
+		out = regexp.MustCompile(`(?m)_+`).ReplaceAllString(out, "_")
+		// Remove leading/trailing underscore
+		out = regexp.MustCompile(`^_|_$`).ReplaceAllString(out, "")
+		return cty.StringVal(strings.ToLower(out)), nil
+	},
+})
+
+var KebabCaseFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name:             "str",
+			Type:             cty.String,
+			AllowDynamicType: true,
+		},
+	},
+	Type: function.StaticReturnType(cty.String),
+	Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+		in := args[0]
+		out, _ := SnakeCase(in)
+		return cty.StringVal(strings.ReplaceAll(out.AsString(), "_", "-")), nil
+	},
+})
+
+var CamelCaseFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name:             "str",
+			Type:             cty.String,
+			AllowDynamicType: true,
+		},
+	},
+	Type: function.StaticReturnType(cty.String),
+	Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+		in := args[0]
+		snake, _ := SnakeCase(in)
+		words := strings.ReplaceAll(snake.AsString(), "_", " ")
+		pascal := strings.ReplaceAll(strings.Title(words), " ", "")
+
+		if pascal != "" {
+			camel := string(strings.ToLower(pascal)[0]) + pascal[1:]
+			return cty.StringVal(camel), nil
+		}
+		return cty.StringVal(""), nil
+	},
+})
+
 // Replace searches a given string for another given substring,
 // and replaces all occurences with a given replacement string.
 func Replace(str, substr, replace cty.Value) (cty.Value, error) {
 	return ReplaceFunc.Call([]cty.Value{str, substr, replace})
 }
+
+// SnakeCase is a Function that converts a given string to snake_case.
+func SnakeCase(str cty.Value) (cty.Value, error) {
+	return SnakeCaseFunc.Call([]cty.Value{str})
+}
+
+// KebabCase is a Function that converts a given string to kebab-case.
+func KebabCase(str cty.Value) (cty.Value, error) {
+	return KebabCaseFunc.Call([]cty.Value{str})
+}
+
+// KebabCase is a Function that converts a given string to kebab-case.
```

Comment:
Sorry for jumping in, but I just noticed this typo in the comment

```suggestion
// CamelCase is a Function that converts a given string to CamelCase.
```

Reply:
// CamelCase is a Function that converts a given string to **c**amelCase.

Reply:
Fixed, thank you!